redis-snapshot-replicator/functions/shipper.py

```diff
@@ -10,10 +10,7 @@
 print('Loading function')
 
 def byTimestamp(snap):
-    if 'SnapshotCreateTime' in snap:
-        return datetime.datetime.isoformat(snap['SnapshotCreateTime'])
-    else:
-        return datetime.datetime.isoformat(datetime.datetime.now())
+    return datetime.datetime.isoformat(snap['NodeSnapshots'][0]['SnapshotCreateTime'])
```

Comment:
the input of this function is a single snapshot no? why are you getting the first one?

Reply:
yes, for each snapshot `NodeSnapshots` is defined as a list:
```
u'NodeSnapshots': [{u'CacheSize': '8 MB', u'SnapshotCreateTime': datetime.datetime(2020, 2, 11, 4, 4, 21, tzinfo=tzlocal()), u'CacheNodeId': '0001', u'CacheNodeCreateTime': datetime.datetime(2017, 12, 4, 19, 37, 9, 123000, tzinfo=tzlocal())}], u'CacheSubnetGroupName': 'frisket-prod-redis', u'Port': 6379, u'PreferredMaintenanceWindow': 'mon:22:00-mon:23:00', u'CacheNodeType': 'cache.m3.medium'}
```
the previous `snap['SnapshotCreateTime']` actually returns a type error (`list indices must be integers or slices, not str`)

Reply:
I see